src/views/createprojectmodal/CreateProjectModal.tsx

```diff
@@ -0,0 +1,173 @@
+import { k8sCreate, k8sDelete, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Form,
+  FormAlert,
+  Modal,
+  ModalVariant,
+  Tab,
+  Tabs,
+  TabTitleText,
+} from '@patternfly/react-core';
+import ExternalLink from '@utils/components/ExternalLink/ExternalLink';
+import { documentationURLs, getDocumentationURL, isManaged } from '@utils/constants/documentation';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { UserDefinedNetworkModel } from '@utils/models';
+import { getResourceURL } from '@utils/resources/shared';
+
+import DetailsProjectTab from './components/DetailsProjectTab';
+import NetworkTab from './components/NetworkTab';
+import { CreateProjectModalFormState, initialFormState, NETWORK_TYPE } from './constants';
+
+import './CreateProjectModal.scss';
+
+const CreateProjectModal: FC<{
+  closeModal: () => void;
+  onSubmit: (project: K8sResourceCommon) => void;
+}> = ({ closeModal, onSubmit }) => {
+  const { t } = useNetworkingTranslation();
+  const projectsURL = getDocumentationURL(documentationURLs.workingWithProjects);
+
+  const navigate = useNavigate();
+
+  const methods = useForm<CreateProjectModalFormState>({
+    defaultValues: initialFormState,
+    mode: 'all',
+  });
+
+  const {
+    formState: { isSubmitting, isValid },
+    handleSubmit,
+    watch,
+  } = methods;
+
+  const project = watch('project');
+  const udn = watch('udn');
+  const networkType = watch('networkType');
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectedTab, setSelectedTab] = useState(0);
+
+  const create = async () => {
+    try {
+      const projectCreated = await k8sCreate({ data: project, model: ProjectRequestModel });
+
+      if (networkType === NETWORK_TYPE.UDN)
+        await k8sCreate({ data: udn, model: UserDefinedNetworkModel }).catch((err) => {
+          k8sDelete({ model: ProjectModel, resource: projectCreated });
+          throw err;
+        });
+
+      closeModal();
+
+      if (onSubmit) {
+        onSubmit(projectCreated);
+      } else {
+        navigate(getResourceURL({ model: ProjectModel, resource: projectCreated }));
+      }
+
+      setErrorMessage('');
+    } catch (error) {
+      setErrorMessage(error.message || t('An error occurred. Please try again.'));
+      // eslint-disable-next-line no-console
+      console.error(`Failed to create Project:`, error);
+    }
+  };
+
+  return (
+    <Modal
+      actions={[
+        <Button
+          data-test="confirm-action"
+          form="create-project-modal-form"
+          id="confirm-action"
+          isDisabled={isSubmitting || !isValid}
+          isLoading={isSubmitting}
+          key="submit"
+          type="submit"
+          variant={ButtonVariant.primary}
+        >
+          {t('Create')}
+        </Button>,
+        <Button
+          data-test-id="modal-cancel-action"
```

Comment:
here you are using data-test-id and the button above is using data-test, which is the correct one? if data-test-id is the correct one, I saw many occurrences across the PR of data-test, please go over all of them and change them